### PR TITLE
Host filter lookup fix

### PR DESCRIPTION
**Pull request by keithjgrant**
##### SUMMARY
* Fixes Host Filter Lookup to use icontains search for name & group fields
* Adds advanced search key sot Host Filter Lookup

addresses #7952

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

**Comment by keithjgrant** [comment]
~~TODO: fix the chip preview when the lookup is closed after selection~~

**Review by jlmitch5 — APPROVED** [review]
looks good (there is a test or linting failing it seems)

File: awx/ui_next/src/components/Lookup/HostFilterLookup.jsx [hunk 1]
```diff
@@ -124,27 +124,38 @@ function HostFilterLookup({
...
+          ),
+          HostsAPI.readOptions(),
+        ]);
         return {
           count: data.count,
           hosts: data.results,
+          relatedSearchableKeys: (
+            actions?.related_search_fields || []
+          ).map(val => val.slice(0, -8)),
+          searchableKeys: Object.keys(actions?.actions.GET || {}).filter(
```
Comment (marshmalien) [inline awx/ui_next/src/components/Lookup/HostFilterLookup.jsx:147]: 🔥

**Review by marshmalien — APPROVED** [review]

**Comment by keithjgrant** [comment]
Now addresses #6801 as well

**Review by tiagodread — APPROVED** [review]
👍🏽 

stubs in:
```
smart-inventory-crud.spec.js
smart-inventory-operations.spec.js
smart-inventory-tabs.spec.js
```

**Pull request merged**